src/sentry/models/release.py

```diff
@@ -214,6 +219,73 @@ def add_project(self, project):
         else:
             return True
 
+    def set_head_commits(self, head_commits, user, fetch_commits=False):
+        from sentry.models import Commit, ReleaseHeadCommit, Repository
+        from sentry.plugins import bindings
+
+        prev_release = type(self).objects.filter(
+            organization_id=self.organization_id,
+            projects__in=self.projects.all(),
+        ).order_by('-date_added').first()
+
+        commit_list = []
+
+        for head_commit in head_commits:
+            repo = Repository.objects.get_or_create(
```

Comment:
Is there a reason we're trying to create a Repository here? It seems, iirc, that we can't reliably do that? We don't really have enough information to fill in. Specifically provider and extenal_id. I feel like we should require the Repository to already exist since this is an action you must do through the UI.